src/EditorFeatures/Core/Implementation/CommentSelection/ToggleBlockCommentCommandHandler.cs

```diff
@@ -0,0 +1,456 @@
+            : base(undoHistoryRegistry, editorOperationsFactoryService)
+        {
+        }
+
+        /* TODO - modify once the toggle block comment handler is added.
+        public VSCommanding.CommandState GetCommandState(CommentSelectionCommandArgs args)
+        {
+            return GetCommandState(args.SubjectBuffer);
+        }
+
+        public bool ExecuteCommand(CommentSelectionCommandArgs args, CommandExecutionContext context)
+        {
+            return ExecuteCommand(args.TextView, args.SubjectBuffer, ValueTuple.Create(), context);
+        }*/
+
+        public override string DisplayName => EditorFeaturesResources.Toggle_Block_Comment;
+
+        protected override string GetTitle(ValueTuple command) => EditorFeaturesResources.Toggle_Block_Comment;
+
+        protected override string GetMessage(ValueTuple command) => EditorFeaturesResources.Toggling_block_comment;
+
+        internal async override Task<CommentSelectionResult> CollectEdits(Document document, ICommentSelectionService service,
+            NormalizedSnapshotSpanCollection selectedSpans, ValueTuple command, CancellationToken cancellationToken)
+        {
+            var emptyResult = new CommentSelectionResult(new List<TextChange>(), new List<CommentTrackingSpan>(), Operation.Uncomment);
+
+            var experimentationService = document.Project.Solution.Workspace.Services.GetRequiredService<IExperimentationService>();
+            if (!experimentationService.IsExperimentEnabled(WellKnownExperimentNames.RoslynToggleBlockComment))
+            {
+                return emptyResult;
+            }
+
+            var commentInfo = await service.GetInfoAsync(document, selectedSpans.First().Span.ToTextSpan(), cancellationToken).ConfigureAwait(false);
+            if (commentInfo.SupportsBlockComment)
+            {
+                return await ToggleBlockComments(document, commentInfo, selectedSpans, cancellationToken).ConfigureAwait(false);
+            }
+
+            return emptyResult;
+        }
+
+        /// <summary>
+        /// Retrieves data about the commented selection that can be optionally overriden by subclasses.
+        /// </summary>
+        protected virtual async Task<IToggleBlockCommentDocumentDataProvider> GetBlockCommentDocumentData(Document document, ITextSnapshot snapshot,
+            CommentSelectionInfo commentInfo, CancellationToken cancellationToken)
+        {
+            return new ToggleBlockCommentDocumentDataProvider(snapshot, commentInfo);
+        }
+
+        private async Task<CommentSelectionResult> ToggleBlockComments(Document document, CommentSelectionInfo commentInfo,
+            NormalizedSnapshotSpanCollection selectedSpans, CancellationToken cancellationToken)
+        {
+            var blockCommentDataProvider = await GetBlockCommentDocumentData(document, selectedSpans.First().Snapshot, commentInfo, cancellationToken).ConfigureAwait(false);
```

Comment:
variable doesn't match method name. are you getting the data? or the data-provider?